api/v1alpha1/containerimagebuild_types.go

```diff
@@ -120,31 +121,42 @@ type ContainerImageBuildSpec struct {
 	PluginData map[string]string `json:"pluginData"`
 }
 
+// ContainerImageBuildState defines a previously observed state of ContainerImageBuild
+type ContainerImageBuildState struct {
+	State           BuildState `json:"state,omitempty"`
+	DurationSeconds string     `json:"durationSeconds,omitempty"`
+}
+
 // ContainerImageBuildStatus defines the observed state of ContainerImageBuild
 type ContainerImageBuildStatus struct {
-	PreviousState    BuildState   `json:"-"` // NOTE: should we persist this value?
-	State            BuildState   `json:"state,omitempty"`
-	ImageURLs        []string     `json:"imageURLs,omitempty"`
-	ErrorMessage     string       `json:"errorMessage,omitempty"`
-	BuildStartedAt   *metav1.Time `json:"buildStartedAt,omitempty"`
-	BuildCompletedAt *metav1.Time `json:"buildCompletedAt,omitempty"`
+	State            BuildState                 `json:"state,omitempty"`
+	States           []ContainerImageBuildState `json:"states,omitempty"`
+	ImageURLs        []string                   `json:"imageURLs,omitempty"`
+	ImageSize        uint64                     `json:"imageSize,omitempty"`
```

Comment:
Just curious, did we want to surface the image size for some reason? I don't think this is bad just wondering is all.

Reply:
I was hoping to report it from the controller, but ended up not doing. Seemed like a useful thing in general.